### PR TITLE
Fix size_t/int mismatch in cms_ec.c and rsa_sig.c

### LINKED ISSUE #20435: size_t vs int confusion in recent commits
PRs #20250 and #20181 correctly pointed out that `RSA_public_decrypt()` and `CMS_SharedInfo_encode()` return `<= 0`  on error, so error checks doing `== 0` are incorrect.

The checks were changed from `== 0` to `<= 0`, however the variables `rslen` and `penclen` are of type `size_t`, so the correct fix would be to assign to an int, check that int for `<= 0` and only if that is not the case assign to the `size_t`.

### PR DESCRIPTION
**Pull request by t8m**
Fixes #20435

### CODE REVIEW

File: crypto/cms/cms_ec.c [hunk 1]
```diff
@@ -358,7 +362,7 @@ static int ecdh_cms_encrypt(CMS_RecipientInfo *ri)
      * of another AlgorithmIdentifier.
      */
     penclen = i2d_X509_ALGOR(wrap_alg, &penc);
-    if (penc == NULL || penclen == 0)
+    if (penc == NULL || penclen <= 0)
```
Comment (botovq) [inline crypto/cms/cms_ec.c:365]: The `NULL` check is redundant (see `asn1_item_flags_i2d()`) and the same strange check appears in `cms_dh.c`, so if you change it here, it should probably be changed there, too.

Comment (t8m) [reply crypto/cms/cms_ec.c]: OK

**Review by paulidale — APPROVED** [review]

**Review by slontis — APPROVED** [review]

**Comment by paulidale** [comment]
Merged with manual fix for 3.1 and 3.0.

**Comment by mattcaswell** [comment]
This seems to have broken the build on 3.1. Possibly due to a bad merge??

```` C
crypto/cms/cms_ec.c:278:16: error: unused variable 'enckeylen' [-Werror,-Wunused-variable]
        size_t enckeylen;
````

**Comment by t8m** [comment]
@mattcaswell Yes, the PR #20516 fixes the bad merge.

**Pull request closed without merging**